### PR TITLE
MCSC-448: Create Date of Death Page

### PR DESCRIPTION
**Pull request by James-Rourke**

### CODE REVIEW

File: conf/messages.cy [hunk 1]
```diff
@@ -397,3 +397,13 @@ pensionSchemeMemberInternationalAddress.townOrCity.change.hidden = cy: Town or c
...
+
+memberDateOfDeath.title = cy: What was their date of death?
+memberDateOfDeath.heading = cy:  What was their date of death?
+memberDateOfDeath.hint = cy: For example, 28 3 1965
+memberDateOfDeath.checkYourAnswersLabel = cy: What was their date of death?
+memberDateOfDeath.error.required.all = cy: Enter the member's date of death
+memberDateOfDeath.error.required.two = cy:The member's date of death must include {0} and {1}
+memberDateOfDeath.error.required = cy: The member's date of death must include {0}
+memberDateOfDeath.error.invalid = cy: Enter the member's date of death in the correct format
+memberDateOfDeath.change.hidden = cy: What was their date of death?
```
Comment (bharti5verma) [inline conf/messages.cy:409]: End of line missing

File: test-utils/generators/UserAnswersGenerator.scala [hunk 2]
```diff
@@ -31,6 +31,8 @@ trait UserAnswersGenerator extends TryValues {
       arbitrary[(PensionSchemeMemberUKAddressPage.type, JsValue)] ::
       arbitrary[(PensionSchemeMemberResidencePage.type, JsValue)] ::
       arbitrary[(WhichPensionSchemeWillPayTaxReliefPage.type, JsValue)] ::
+      arbitrary[(MemberDateOfDeathPage.type, JsValue)] ::
+      arbitrary[(WhichPensionSchemeWillPayTaxReliefPage.type, JsValue)] ::
```
Comment (bharti5verma) [inline test-utils/generators/UserAnswersGenerator.scala:35]: "WhichPensionSchemeWillPayTaxReliefPage" Is this duplicated here.

File: test/pages/PensionSchemeMemberDOBPageSpec.scala [hunk 3]
```diff
@@ -38,16 +38,36 @@ class PensionSchemeMemberDOBPageSpec extends PageBehaviours {
...
+    val userAnswers = emptyUserAnswers
+      .set(page, LocalDate.of(1995, 1, 1))
+      .success
+      .value
+
+    if (status == status.PowerOfAttorney) {
+      val nextPageUrl: String = page.navigate(NormalMode, userAnswers).url
+
+      checkNavigation(nextPageUrl, "/their-nino")
+    }
```
Comment (bharti5verma) [inline test/pages/PensionSchemeMemberDOBPageSpec.scala:70]: Could you please add Summary spec file as well.

File: conf/messages.cy (hunk 1 above)
Comment (bharti5verma) [reply conf/messages.cy]: Fixed.

File: test/pages/PensionSchemeMemberDOBPageSpec.scala (hunk 3 above)
Comment (bharti5verma) [reply test/pages/PensionSchemeMemberDOBPageSpec.scala:70]: Added.

**Review by ElliotAndersonHMRC — APPROVED** [review]
looks good!

**Pull request merged**